libs/authorizenet/src/reducers/authorize-net/authorize-net-reducer.interface.ts

```diff
@@ -1,6 +1,5 @@
-import { DaffAuthorizeNetTokenResponse } from '../../models/response/authorize-net-token-response';
-
-export interface DaffAuthorizeNetReducerState<T extends DaffAuthorizeNetTokenResponse> {
-	tokenResponse: T;
+export interface DaffAuthorizeNetReducerState {
+	cc_last_4: string;
```

Comment:
I'm confused as to what is going on with `cc_last_4`. Why is it stored here and not included with the cart payload?

Reply:
The only reason this is here is because it is needed for adding an authorizeNet payment method to the cart, and it is specific to the authorizeNet payment method. Really, I think there should be no reason why the last 4 digits are needed for placing the order. You'd think that the token would be all that is needed. But apparently authorizeNet needs the last four digits too. So, the cart package can't really be generic with all payment methods and know the last 4 digits, because the `DaffCartPaymentMethod.payment_info` is a completely custom thing that is built by the payment method module. It's easier to see what I mean if you compare magento payment methods. For authorizeNet, the data needed to set the payment method on a magento cart looks like this:
```
code: "authorizenet_acceptjs"
authorizenet_acceptjs: {
   cc_last_4: 1111
   opaque_data_descriptor: "COMMON.ACCEPT.INAPP.PAYMENT"
   opaque_data_value: "<nonce_value>"
}
```
One of the paypal payment methods looks like this instead:
```
code: "payflow_advanced"
payflow_link: {
   return_url: "paypal/action/return.html"
   error_url: "paypal/action/error.html"
   cancel_url: "paypal/action/cancel.html"
}
```
So instead of having the cart module know any of that random stuff, I put it in the payment methods

Reply:
I don't see a reason for keeping `cc_last_4` in state. It seems that the only place its used is in the `generateToken$` effect, which has access to the CC number through the action payload. I'd say don't bother keeping it in state unless it will be needed in places that won't have the generate token action available. I could be missing something though.

Reply:
Yea, you're totally right. I can't remember why I even put it there in the first place. Removed